websites/A/Anime4up/presence.ts

```diff
@@ -1,22 +1,161 @@
+  run?(): PresenceData;
+}
+
+enum Settings {
+  TIMESTAMP = "timestamp",
+  BUTTONS = "buttons",
+  LOGO = "logo"
+}
+
+enum Icons {
+  PAUSED = "paused",
+  PLAYED = "played",
+  SEARCHING = "searching",
+  LOCATION = "location",
+  DISCOVERY = "discovery"
+}
+
+enum Logos {
+  LIGHT = "light-logo",
+  DARK = "dark-logo"
 }
 
+let video = { duration: 0, currentTime: 0, paused: true };
+
+const presence = new Presence({
+    clientId: "770030754356396052"
+  }),
+  startTimestamp: number = Math.floor(Date.now() / 1000),
+  router = ({ path, data }: { path: string; data: PresenceData }): Route => {
+    const routes: Route[] = [
+      { path: /^\/$/, details: () => "On Homepage" },
+      {
+        path: /^\/episode\//,
+        run: () => {
+          const timestamps = presence.getTimestamps(
+            Math.floor(video.currentTime),
+            Math.floor(video.duration)
+          );
+
+          [, data.endTimestamp] = timestamps;
+
+          if (video.paused) {
+            delete data.startTimestamp;
+            delete data.endTimestamp;
+          }
+
+          return data;
+        },
+
+        playback: () => !video.paused,
+
+        smallImageKey: () => (video.paused ? Icons.PAUSED : Icons.PLAYED),
+
+        smallImageText: () => (video.paused ? "Paused" : "Played"),
+
+        state: () =>
+          document
+            .querySelectorAll(".container")[1]
+            .textContent.slice(1, -1)
+            .split(" ")
+            .slice(0, -2)
+            .join(" "),
+
+        details: () =>
+          `Episode: ${document
+            .querySelectorAll(".container")[1]
+            .textContent.slice(1, -1)
+            .split(" ")
+            .pop()}`,
+
+        buttons: () => [
+          { label: "Watch Episode", url: location.href },
+          {
+            label: "Anime Page",
+            url: document
+              .querySelector(".anime-page-link")
+              .querySelector("a")
+              .getAttribute("href")
+          }
+        ]
+      },
+      {
+        path: /^\/\?search_param=(.*)/,
+
+        smallImageKey: () => Icons.SEARCHING,
+
+        smallImageText: () => "Searching",
+
+        details: () =>
+          `Results: ${document.querySelectorAll(".col-lg-2").length ?? 0}`,
+
+        state: () =>
+          `Searching: ${document
+            .querySelectorAll(".container")[1]
+            .textContent.split(" ")
+            .slice(4, -1)
+            .join(" ")}`,
+
+        buttons: () => [{ label: "Results", url: location.href }]
+      },
+      {
+        path: /^\/anime\/(.*)/,
+
+        smallImageKey: () => Icons.LOCATION,
+
+        smallImageText: () => "Viewing",
+
+        details: () => "Viewing an Anime",
+
+        state: () => document.querySelector(".anime-details-title").textContent,
+
+        buttons: () => [
+          { label: "View Anime", url: location.href },
+          {
+            label: "Last Episode",
+            // prettier-ignore
```

Comment:
no prettier ignore 😠

Reply:
unfortunately, I've to put it, for some reason when I format the file via prettier, eslint will suddenly throw an error

Reply:
What eslint throws then?

Reply:
![image](https://user-images.githubusercontent.com/59054736/125203120-d78d0780-e27f-11eb-8632-385b6f834a59.png)

Unexpected newline between object and [ of property access.eslintno-unexpected-multiline

Reply:
Then try assigning this value to variable bc it looks like parser issue from eslint

Reply:
and the same thing for GMANGA presence right?

Reply:
This is in GMANGA too? Well yeah then